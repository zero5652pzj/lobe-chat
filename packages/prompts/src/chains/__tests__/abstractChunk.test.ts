import { describe, expect, it, vi } from 'vitest';

import { chainAbstractChunkText } from '../abstractChunk';

describe('chainAbstractChunkText', () => {
  it('should generate correct chat payload for chunk text', () => {
    const testText = 'This is a sample chunk of text that needs to be summarized.';

    const result = chainAbstractChunkText(testText);

    expect(result).toMatchSnapshot();
  });

  it('should handle text with special characters', () => {
    const testText = 'Text with special chars: @#$%^&*()';

    const result = chainAbstractChunkText(testText);

    expect(result.messages![1].content).toBe(testText);
  });

  it('should always use system role for first message', () => {
    const result = chainAbstractChunkText('test');

    expect(result.messages![0].role).toBe('system');
  });

  it('should always use user role for second message', () => {
    const result = chainAbstractChunkText('test');

    expect(result.messages![1].role).toBe('user');
  });
});
