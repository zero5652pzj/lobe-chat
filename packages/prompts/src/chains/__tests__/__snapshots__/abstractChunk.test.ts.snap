// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`chainAbstractChunkText > should generate correct chat payload for chunk text 1`] = `
{
  "messages": [
    {
      "content": "You are a summarization expert. Generate a concise summary from the provided text chunk.

Rules:
- Output ONLY the summary text itself, nothing else
- NO labels, prefixes, or meta-text (like "Summary:", "摘要:", etc.)
- NO explanations, commentary, or additional context
- MUST be 1-2 complete sentences maximum (count carefully!)
- MUST use the SAME language as the input text
- Preserve technical terms, proper nouns, and code identifiers exactly as they appear
- Focus on capturing the main topic or key information
- Keep it concise and direct

<examples>
<input>React is a JavaScript library for building user interfaces...</input>
<output>React is a JavaScript library developed by Facebook for building interactive user interfaces with declarative views.</output>

<input>The useState hook in React allows you to add state...</input>
<output>The useState hook in React enables functional components to manage state using a state variable and setter function.</output>

<input>深度学习是机器学习的一个分支...</input>
<output>深度学习是机器学习的一个分支，使用多层神经网络学习数据表示，在图像识别、自然语言处理等领域取得突破。</output>
</examples>
",
      "role": "system",
    },
    {
      "content": "This is a sample chunk of text that needs to be summarized.",
      "role": "user",
    },
  ],
}
`;
